src/components/Editor/MonacoInstance/actions.ts

```diff
@@ -155,18 +155,26 @@ const setupCustomActions = (
     actions: EditorAction[],
     onCustomAction: CustomActionCallback
 ) => {
-    actions.forEach(({actionName, label, keyBindings}) => {
+    actions.forEach(({actionName, label, keyBindings, contextMenuGroupId, contextMenuOrder}) => {
         if (null === actionName) {
             return;
         }
-        editor.addAction({
+        const descriptor: monaco.editor.IActionDescriptor = {
             id: actionName,
             label: label,
             keybindings: keyBindings,
             run: () => {
                 onCustomAction(editor, actionName);
             },
-        });
+        };
+
+        if (null !== contextMenuGroupId) {
+            descriptor.contextMenuGroupId = contextMenuGroupId;
+        }
+        if (null !== contextMenuOrder) {
+            descriptor.contextMenuOrder = contextMenuOrder;
+        }
+        editor.addAction(descriptor);
     });
```

Comment:
another one (please discard https://github.com/y-scope/yscope-log-viewer/pull/197/files#r1980674897 if you agree with below): if we are keeping `contextMenuGroupId` and `contextMenuOrder` optional as discussed at https://github.com/y-scope/yscope-log-viewer/pull/197/files#r1981893238 , we can write

```suggestion
    actions.forEach(({actionName, label, keyBindings, ...rest}) => {
        if (null === actionName) {
            return;
        }
        editor.addAction({
            id: actionName,
            keybindings: keyBindings,
            label: label,
            run: () => {
                onCustomAction(editor, actionName);
            },
            ...rest,
        });
    });
```